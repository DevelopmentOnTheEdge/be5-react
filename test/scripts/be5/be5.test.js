import React from 'react';
import be5               from '../../../src/scripts/be5/be5';
import {getRoute, registerRoute} from '../../../src/scripts/be5/core/routes';


test('be5.locale', () => {
  expect(be5.locale.msg('settings')).toBe('Settings');

  be5.locale.addMessages('en', {'test':'Test message'});

  expect(be5.locale.msg('test')).toBe('Test message');
});

test('be5.url', () =>
{
  expect(be5.url.empty()).toBe(true);
  be5.url.clear();
  be5.url.set('text/Test');

  expect(be5.url.create(['form', 'users','All records', 'Insert'], {'user_name':'Guest'}))
    .toBe('form/users/All records/Insert/user_name=Guest');

  expect(be5.url.parse('form/users/All records/Insert/user_name=Guest'))
    .toEqual({ positional:['form', 'users','All records', 'Insert'], named: {'user_name':'Guest'} });

  const action = jest.fn();

  registerRoute('form', action);
  expect(getRoute('form')).toBe(action);

  be5.url.process('testDoc', '#!form/users/All records/user_name=Guest');

  expect(action.mock.calls.length).toBe(1);
  expect(action.mock.calls[0]).toEqual(["testDoc", "users", "All records", {"user_name": "Guest"}]);
});

