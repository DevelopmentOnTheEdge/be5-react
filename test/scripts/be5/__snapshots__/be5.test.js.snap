// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="SplitPane vertical"
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "left": 0,
      "outline": "none",
      "overflow": "hidden",
      "position": "absolute",
      "right": 0,
    }
  }
>
  <div
    className="Pane vertical"
    style={
      Object {
        "flex": "none",
        "outline": "none",
        "overflow": "auto",
        "position": "relative",
        "width": 280,
      }
    }
  >
    <div
      className="side"
    >
      <div
        className="menuContainer"
      >
        <div
          className="menuHeader"
        >
          <h2>
            Hello!
          </h2>
        </div>
        <input
          className="searchField form-control"
          onChange={[Function]}
          placeholder="Filter..."
          type="text"
          value=""
        />
        <div
          className="menu"
        >
          <div
            className="menuNode"
          >
            <span
              className="rootMenuItem menuItemWithoutRef"
            >
              Loading...
            </span>
            <div />
          </div>
        </div>
        <div
          className="menuFooter"
        >
          <img
            src={Object {}}
          />
        </div>
      </div>
      <hr />
      <h3>
        Settings
      </h3>
      <div
        className="languageBox"
      >
        <div
          className="languageList"
        />
      </div>
      <div
        className="roleBox"
      />
    </div>
  </div>
  <span
    className="Resizer vertical"
    onMouseDown={[Function]}
  />
  <div
    className="Pane vertical"
    style={
      Object {
        "flex": 1,
        "outline": "none",
        "overflow": "auto",
        "position": "relative",
      }
    }
  >
    <div
      className="content"
    >
      <h1
        className="text-center"
      >
        Page is loading...
      </h1>
    </div>
  </div>
</div>
`;

exports[`snapshot 2`] = `
<div
  className="SplitPane vertical"
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "height": "100%",
      "left": 0,
      "outline": "none",
      "overflow": "hidden",
      "position": "absolute",
      "right": 0,
    }
  }
>
  <div
    className="Pane vertical"
    style={
      Object {
        "flex": "none",
        "outline": "none",
        "overflow": "auto",
        "position": "relative",
        "width": 280,
      }
    }
  >
    <div
      className="side"
    >
      <div
        className="menuContainer"
      >
        <div
          className="menuHeader"
        >
          <h2>
            Hello!
          </h2>
        </div>
        <input
          className="searchField form-control"
          onChange={[Function]}
          placeholder="Filter..."
          type="text"
          value=""
        />
        <div
          className="menu"
        >
          <div
            className="menuNode"
          >
            <span
              className="rootMenuItem menuItemWithoutRef"
            >
              Loading...
            </span>
            <div />
          </div>
        </div>
        <div
          className="menuFooter"
        >
          <img
            src={Object {}}
          />
        </div>
      </div>
      <hr />
      <h3>
        Settings
      </h3>
      <div
        className="languageBox"
      >
        <div
          className="languageList"
        />
      </div>
      <div
        className="roleBox"
      />
    </div>
  </div>
  <span
    className="Resizer vertical"
    onMouseDown={[Function]}
  />
  <div
    className="Pane vertical"
    style={
      Object {
        "flex": 1,
        "outline": "none",
        "overflow": "auto",
        "position": "relative",
      }
    }
  >
    <div
      className="documentContent"
    >
      <div
        className="staticPage"
        dangerouslySetInnerHTML={
          Object {
            "__html": "test",
          }
        }
      />
    </div>
  </div>
</div>
`;
